de.metas.swat/de.metas.swat.base/src/main/java/de/metas/impex/api/impl/InputDataSourceDAO.java

```diff
@@ -33,16 +35,18 @@
 
 import de.metas.cache.annotation.CacheCtx;
 import de.metas.cache.annotation.CacheTrx;
+import de.metas.impex.InputDataSourceId;
 import de.metas.impex.api.IInputDataSourceDAO;
 import de.metas.impex.api.InputDataSourceCreateRequest;
 import de.metas.impex.model.I_AD_InputDataSource;
 import de.metas.util.Services;
+import de.metas.util.lang.ExternalId;
 import lombok.NonNull;
 
 public class InputDataSourceDAO implements IInputDataSourceDAO
 {
 	@Override
-	public int retrieveInputDataSourceId(final String internalName)
+	public int retrieveInputDataSourceIdByInternalName(final String internalName)
```

Comment:
introduce a InputDataSourceQuery, similar to, `de.metas.bpartner.service.BPartnerQuery` (i think InputDataSourceQuery can be simpler)

Reply:
done